packages/console/src/pages/EnterpriseSsoDetails/Connection/index.tsx

```diff
@@ -0,0 +1,144 @@
+import { useEffect } from 'react';
+import { useForm, FormProvider } from 'react-hook-form';
+import { toast } from 'react-hot-toast';
+import { useTranslation } from 'react-i18next';
+
+import DetailsForm from '@/components/DetailsForm';
+import FormCard from '@/components/FormCard';
+import UnsavedChangesAlertModal from '@/components/UnsavedChangesAlertModal';
+import useApi from '@/hooks/use-api';
+import BasicInfo from '@/pages/EnterpriseSso/Guide/BasicInfo';
+import OidcMetadataForm from '@/pages/EnterpriseSso/Guide/OidcMetadataForm';
+import SamlAttributeMapping from '@/pages/EnterpriseSso/Guide/SamlAttributeMapping';
+import SamlMetadataForm from '@/pages/EnterpriseSso/Guide/SamlMetadataForm';
+import {
+  type SsoConnectorWithProviderConfigWithGeneric,
+  type ParsedSsoIdentityProviderConfig,
+  type GuideFormType,
+  type SsoConnectorConfig,
+} from '@/pages/EnterpriseSso/types';
+import { trySubmitSafe } from '@/utils/form';
+
+import * as styles from './index.module.scss';
+
+type Props<T extends SsoProviderName> = {
+  isDeleted: boolean;
+  data: SsoConnectorWithProviderConfigWithGeneric<T>;
+  onUpdated: (data: SsoConnectorWithProviderConfigWithGeneric<T>) => void;
+};
+
+// This component contains only `data.config` and `data.providerProperties`.
+function Connection<T extends SsoProviderName>({ isDeleted, data, onUpdated }: Props<T>) {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const {
+    id: ssoConnectorId,
+    connectorName: ssoConnectorName,
+    providerName,
+    providerProperties,
+    providerConfig,
+    config,
+  } = data;
+
+  const api = useApi();
+
+  const methods = useForm<GuideFormType<T>>();
+
+  const {
+    formState: { isSubmitting, isDirty },
+    handleSubmit,
+    reset,
+  } = methods;
+
+  useEffect(() => {
+    reset(config);
+  }, [config, reset]);
+
+  const onSubmit = handleSubmit(
+    trySubmitSafe(async (formData) => {
+      if (isSubmitting) {
+        return;
+      }
+
+      const updatedSsoConnector = await api
+        .patch(`api/sso-connectors/${ssoConnectorId}/config`, {
+          json: cleanDeep(formData),
+          searchParams: new URLSearchParams({ shouldReplace: 'true' }),
+        })
+        .json<SsoConnectorWithProviderConfigWithGeneric<T>>();
+
+      toast.success(t('general.saved'));
+      onUpdated(updatedSsoConnector);
+    })
+  );
+
+  return (
+    <FormProvider {...methods}>
+      <DetailsForm
+        isDirty={isDirty}
+        isSubmitting={isSubmitting}
+        onDiscard={reset}
+        onSubmit={onSubmit}
+      >
+        <FormCard
+          title="enterprise_sso_details.upload_idp_metadata_title"
+          description="enterprise_sso_details.upload_idp_metadata_description"
+        >
+          {providerName === SsoProviderName.OIDC ? (
+            // Can not infer the type by narrowing down the value of `providerName`, so we need to cast it.
+            <OidcMetadataForm
+              isGuidePage={false}
```

Comment:
Seems `isGuidePage` is optional, so we don't need to pass `false` to it

Reply:
Intentially declare this for better readability to show that this is not on the guide page.